### PR TITLE
Feat: Test for entities

### PR DESCRIPTION
**Pull request by malik672**
this test is for entities

## Summary by CodeRabbit

- **Tests**
	- Introduced new test cases for the Trade module, enhancing validation of trading scenarios involving Ether and other tokens.
	- Added tests for trade construction, ensuring accuracy of input/output currency handling and edge case management.
	- Validated methods for finding the best trading routes, confirming functionality under various liquidity conditions.
	- Ensured reliable output for maximum amount in, minimum amount out, and worst execution price methods through comprehensive testing.
	- Created a dedicated testing module for improved organization and modularity of entity-related tests.

### CODE REVIEW

File: src/entities/entities_test.rs [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        )
+    });
+    static DECIMAL_PERMUTATIONS: [[u8; 3]; 3] = [[0, 0, 0], [0, 9, 18], [18, 18, 18]];
+
+    fn decimalize(amount: u128, decimals: u8) -> BigInt {
+        BigInt::from(amount) * BigInt::from(10).pow(decimals as u32)
+    }
+
+    #[test]
+    fn test_entities() {
```
Comment (shuhuiluo) [inline src/entities/entities_test.rs:29]: Why don't you use modules and functions to scope unit tests? How is this a unit test?

**Review by shuhuiluo — CHANGES_REQUESTED** [review]
